File: lib/samples/nifty/start_screen_controller.rb
java_import "com.jme3.app.Application"
java_import "com.jme3.app.SimpleApplication"
java_import "com.jme3.app.state.AbstractAppState"
java_import "com.jme3.app.state.AppStateManager"
java_import "de.lessvoid.nifty.Nifty"
java_import "de.lessvoid.nifty.screen.Screen"
java_import "de.lessvoid.nifty.screen.ScreenController"

class StartScreenController < AbstractAppState
  include ScreenController
  field_accessor :initialized
  
  # Java method not found: com.jme3.app.state.AbstractAppState.initialize()
  #java_alias :init_with, :initialize
  
  attr_accessor :nifty, :screen, :app
  
  def initialize
  end
  
  def init_with(data = {})
    #init = java_method(:initialize, [com.jme3.app.state.AppStateManager, com.jme3.app.SimpleApplication])
    #init.call(data[:state_manager], data[:app])
    @state_manager = data[:state_manager]
    self.app = data[:app]
    self.initialized = true
    return self # mocking a "initialize" method
  end
  
  def bind(nifty, screen)
    puts "\n\n BINDING \n\n"
    self.nifty = nifty
    self.screen = screen
  end
  
  def onStartScreen
    puts "\n\n on_start_screen called: #{initialized}\n\n"
  end
  
  def onEndScreen
    puts "\n\n on_end_screen called\n\n"
  end
  
  def update(tpf)
    puts "\n\n UPDATING\n\n"
    # jme update loop
  end
  
  def startGame(next_screen)
    puts "\n\n Called start for #{next_screen}\n\n"
    nifty.goto_screen(next_screen)
  end
  
  def quitGame()
    puts "\n\n Called quit\n\n"
    app.stop
  end
  
end

File: lib/samples/nifty/StartScreenController.java
import com.jme3.app.Application;
import com.jme3.app.SimpleApplication;
import com.jme3.app.state.AbstractAppState;
import com.jme3.app.state.AppStateManager;
import de.lessvoid.nifty.Nifty;
import de.lessvoid.nifty.screen.Screen;
import de.lessvoid.nifty.screen.ScreenController;
 
public class StartScreenController extends AbstractAppState implements ScreenController {
 
  private Nifty nifty;
  private Screen screen;
  private SimpleApplication app;
 
  /** custom methods */ 
 
  public StartScreenController(Application app) { 
    /** You custom constructor, can accept arguments */
    this.app = (SimpleApplication) app;
  } 
 
  /** Nifty GUI ScreenControl methods */ 
 
  public void bind(Nifty nifty, Screen screen) {
    this.nifty = nifty;
    this.screen = screen;
  }
 
  public void onStartScreen() { }
 
  public void onEndScreen() { }
 
  /** jME3 AppState methods */ 
 
  @Override
  public void initialize(AppStateManager stateManager, Application app) {
    super.initialize(stateManager, app);
  }
 
  @Override
  public void update(float tpf) { 
    /** jME update loop! */ 
  }
  
  public void startGame(String nextScreen) {
    System.out.println("\n\n startGame called\n\n");
    nifty.gotoScreen(nextScreen);  // switch to another screen
    // start the game and do some more stuff...
  }

  public void quitGame() {
    System.out.println("\n\n quitGame called\n\n");
    app.stop();
    // TODO:
    // find app instance variable @controller_path and delete that file.
  }
 
}